lib/typing/decimal/base.go

```diff
@@ -31,18 +31,13 @@ func (d *DecimalDetails) isBigNumeric() bool {
 	}
 
 	// max(1,s) <= p <= s + 38
-	return numbers.BetweenEq(max(1, d.scale), d.scale+38, *d.precision)
+	return numbers.BetweenEq(max(1, d.scale), d.scale+38, d.precision)
 }
 
 func (d *DecimalDetails) toKind(maxPrecision int32, exceededKind string) string {
-	precision := maxPrecision
```

Comment:
There is a subtle change here in that before if `d.precision` was nil we would use the maxPrecision and now we'll treat it the same as `-1`.

Reply:
I think that's fine